homeassistant/components/freebox/base_class.py

```diff
@@ -0,0 +1,197 @@
+        return False
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._id)},
+            "name": self._device_name,
+            "manufacturer": self._manufacturer,
+            "model": self._model,
+            "sw_version": self._firmware,
+        }
+
+    async def async_update_signal(self):
+        """Update signal."""
+        self._node = self._router.home_devices[self._id]
+        # Update NAME
+        if self._sub_node is None:
+            self._name = self._node["label"].strip()
+        else:
+            self._name = (
+                self._node["label"].strip() + " " + self._sub_node["label"].strip()
+            )
+        self.async_write_ha_state()
+
+    async def set_home_endpoint_value(self, command_id, value=None):
+        """Set Home endpoint value."""
+        if value is None:
+            value = {"value": None}
+        if command_id == VALUE_NOT_SET:
+            _LOGGER.error(
+                "Unable to SET a value through the API. Command is VALUE_NOT_SET"
+            )
+            return False
+        await self._router.api.home.set_home_endpoint_value(self._id, command_id, value)
+        return True
+
+    async def get_home_endpoint_value(self, command_id):
+        """Get Home endpoint value."""
+        if command_id == VALUE_NOT_SET:
+            _LOGGER.error(
+                "Unable to GET a value through the API. Command is VALUE_NOT_SET"
+            )
+            return VALUE_NOT_SET
+        try:
+            node = await self._router.api.home.get_home_endpoint_value(
+                self._id, command_id
+            )
+        except TimeoutError:
+            _LOGGER.warning("The Freebox API Timeout during a value retrieval")
+            return VALUE_NOT_SET
+        return node.get("value", VALUE_NOT_SET)
+
+    def get_command_id(self, nodes, ep_type, name):
+        """Get the command id."""
+        node = next(
+            filter(lambda x: (x["name"] == name and x["ep_type"] == ep_type), nodes),
+            None,
+        )
+        if node is None:
+            _LOGGER.warning(
+                "The Freebox Home device has no value for: " + ep_type + "/" + name
+            )
+            return VALUE_NOT_SET
+        return node["id"]
+
+    def get_value(self, ep_type, name):
+        """Get the value."""
+        node = next(
+            filter(
+                lambda x: (x["name"] == name and x["ep_type"] == ep_type),
+                self._node["show_endpoints"],
+            ),
+            None,
+        )
+        if node is None:
+            _LOGGER.warning(
+                "The Freebox Home device has no node for: " + ep_type + "/" + name
+            )
+            return VALUE_NOT_SET
+        return node.get("value", VALUE_NOT_SET)
```

Comment:
```suggestion
            return None
        return node.get("value") or None
```
_or None replace "" value by None_

I think you better return `None` instead of `VALUE_NOT_SET`, one is empty/null, the other one is a string and may be interpret differentry by sensors device class or platforms.